README.md

```diff
@@ -14,56 +14,66 @@ such as Libvirt, QEMU and KVM.
-
-```bash
-export TF_VAR_SSH_HOST="XXX.XXX.XXX.XXX"
-export TF_VAR_SSH_USERNAME="yourUsername"
-export TF_VAR_SSH_PASSWORD="yourPassword"
-export TF_VAR_SSH_PRIVATE_KEY_PATH="~/.ssh/id_ed25519"
-```
-
-2. Execute some REQUIRED previous scripts to bootstrap the host
-
-> By the moment, only recent Ubuntu versions are supported. 
-> Feel free to extend the OS support pushing your code to this repository.
-
-```bash
-# Copy current SSH key into the target host
-echo ${TF_VAR_SSH_PASSWORD} | ssh-copy-id -f ${TF_VAR_SSH_USERNAME}@${TF_VAR_SSH_HOST}
-
-# Give execution permissions to the helper scripts
-chmod -R +x ./scripts
-
-# Connect to the host machine by SSH and install the dependencies using passwordless authentication 
-# (user with sudo privileges required) 
-ssh -i ${TF_VAR_SSH_PRIVATE_KEY_PATH} ${TF_VAR_SSH_USERNAME}@${TF_VAR_SSH_HOST} \
-  "sudo bash ./scripts/prepare-host-ubuntu.sh ${TF_VAR_SSH_USERNAME}"
-```
-
-3. Configure Terraform to store the state of your infrastructure
-
-> We have configured S3 backend by default thinking on bare metal solutions like S3-compatible APIs provided by 
-> solutions like TrueNAS, which cover the problem of storing the .tfstate using the same approach as major cloud providers
-
-```bash
-# Set the right parameters for your S3 storage
-nano backends/config.s3.tfbackend
-
-# Init the process configuring your backend
-terraform init -backend-config=backends/config.s3.tfbackend
-```
-
-4. Declare resources related to VMs to create.
-
-> Change the file called `data.tf`.
-> Several examples can be found inside (yes, for Kubernetes)
-
-5. Create your VMs.
-```bash
-terraform apply
-```
+1. Declare environment vars with the SSH connection parameters.
+   These can be declared as input vars inside a `.tfvars` file too
+
+   ```bash
+    export TF_VAR_SSH_HOST="XXX.XXX.XXX.XXX"
+    export TF_VAR_SSH_USERNAME="yourUsername"
+    export TF_VAR_SSH_PASSWORD="yourPassword"
+    export TF_VAR_SSH_PRIVATE_KEY_PATH="~/.ssh/id_ed25519"
+    ```
+
+2. Install some REQUIRED dependencies in local machine
+
+   > To build the ISOs we need to have installed `mkisofs`.
+
+   ```bash
+   sudo apt install mkisofs
+   ```
+
+3. Execute some REQUIRED previous scripts to bootstrap the host
+
+   > By the moment, only recent Ubuntu versions are supported.
+   > Feel free to extend the OS support by pushing your code to this repository.
+
+   ```bash
+   # Copy current SSH key into the target host
+   echo ${TF_VAR_SSH_PASSWORD} | ssh-copy-id -f ${TF_VAR_SSH_USERNAME}@${TF_VAR_SSH_HOST}
+   
+   # Give execution permissions to the helper scripts
+   chmod -R +x ./scripts
+   
+   # Connect to the host machine by SSH and install the dependencies using passwordless authentication 
+   # (user with sudo privileges required) 
+   scp ./scripts/prepare-host-ubuntu.sh ${TF_VAR_SSH_USERNAME}@${TF_VAR_SSH_HOST}:/tmp
+   ssh ${TF_VAR_SSH_USERNAME}@${TF_VAR_SSH_HOST} "sudo bash ./tmp/prepare-host-ubuntu.sh ${TF_VAR_SSH_USERNAME}"
+   ```
```

Comment:
If I follow the previous instructions that line fails because the script doesn’t exist in the remote machine. So we need copy it or do something like:

```
ssh ${TF_VAR_SSH_USERNAME}@${TF_VAR_SSH_HOST} "sudo bash -s " < scripts/prepare-host-ubuntu.sh ${TF_VAR_SSH_USERNAME}"
```

Now I can’t test it.

Reply:
Well, I assume the previous proposal is actually tested, I mean, the proposal you want to merge, right?

Reply:
Yes yes, I did it.

Reply:
Before the last commit, this script was applied using a "remote-exec" provisioner. I think that the provisioner copy the file under the hood.

Reply:
Ok, so we will merge it as it fixes the problem :)